fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java

```diff
@@ -723,6 +724,336 @@ public void updateRecords(@Nonnull Descriptors.FileDescriptor recordsDescriptor)
+            if (messageType == null) {
+                throw new MetaDataException("Record type " + recordType + " does not exist");
+            }
+            DescriptorProtos.FieldDescriptorProto.Builder fieldBuilder = findFieldByName(messageType, field.getName());
+            if (fieldBuilder != null) {
+                throw new MetaDataException("Field " + field.getName() + " already exists in record type " + recordType);
+            }
+            messageType.addField(field);
+            RecordMetaDataBuilder recordMetaDataBuilder = createMetaDataBuilder(builder.build());
+            recordMetaDataBuilder.setVersion(metaDataProto.getVersion() + 1);
+            return saveAndSetCurrent(recordMetaDataBuilder.getRecordMetaData().toProto());
+        });
+    }
+
+    @Nullable
+    private DescriptorProtos.DescriptorProto.Builder findMessageTypeByName(@Nonnull DescriptorProtos.FileDescriptorProto.Builder recordsBuilder, @Nonnull String recordType) {
+        return recordsBuilder.getMessageTypeBuilderList().stream().filter(m -> m.getName().equals(recordType)).findAny().orElse(null);
+    }
+
+    /**
+     * Deprecate a field from a record type.
+     *
+     * @param recordType the record type to deprecate the field from
+     * @param fieldName the name of the field to be deprecated
+     */
+    public void deprecateField(@Nonnull String recordType, @Nonnull String fieldName) {
+        context.asyncToSync(FDBStoreTimer.Waits.WAIT_DEPRECATE_FIELD, deprecateFieldAsync(recordType, fieldName));
+    }
+
+    /**
+     * Deprecate a field from a record type asynchronously.
+     *
+     * @param recordType the record type to deprecate the field from
+     * @param fieldName the name of the field to be deprecated
+     * @return a future that completes when the record type is deprecated
+     */
+    @Nonnull
+    public CompletableFuture<Void> deprecateFieldAsync(@Nonnull String recordType, @Nonnull String fieldName) {
+        return loadCurrentProto().thenCompose(metaDataProto -> {
+            RecordMetaDataProto.MetaData.Builder builder = metaDataProto.toBuilder();
+
+            // Find the record type
+            DescriptorProtos.DescriptorProto.Builder messageType = findMessageTypeByName(builder.getRecordsBuilder(), recordType);
+            if (messageType == null) {
+                throw new MetaDataException("Record type " + recordType + " does not exist");
+            }
+
+            // Deprecate the field
+            DescriptorProtos.FieldDescriptorProto.Builder fieldBuilder = findFieldByName(messageType, fieldName);
+            if (fieldBuilder == null) {
+                throw new MetaDataException("Field " + fieldName + " not found in record type " + recordType);
+            }
+            if (fieldBuilder.hasOptions()) {
+                fieldBuilder.setOptions(fieldBuilder.getOptions().toBuilder().setDeprecated(true).build());
```

Comment:
```suggestion
                fieldBuilder.getOptionsBuilder().setDeprecated(true);
```

It's possible the mechanics of actually deprecating a field once you have a builder (which is currently shared between this and the record deprecation logic) would be worth encapsulating into its own function.